release: setup for post-blockers cmd

celiala opened the pull request:
### Summary

This commit does a quick refactor and sets up cobra command infra for the post-blockers command. Actual logic for fetching and posting blockers to an email will happen in a follow-up commit. The intent here is to a minimal refactor so that remaining "Week -1" and "Week 1" can happen in parallel (and avoid merge conflicts).

### Tests

I ran the `pick-sha` command locally with fake parameters to verify no refactor regressions[See Test 1]. Additionally, I ran the `post-blockers` command locally to get expected ("TODO") output[See Test 2].

#### Test 1:
- Ran `pick-sha`, using fake parameters
email`
- Got error (expected): `panic: cannot find next release: cannot find healthy build: no ref found`

```
SMTP_PASSWORD=foo \
JIRA_USERNAME=foo \
JIRA_TOKEN=foo \
bazel run pkg/cmd/release:release \
pick-sha -- \
  --dry-run \
  --release-series=21.1 \
  --smtp-user=foo \
  --smtp-host=smtp.gmail.com \
  --smtp-port=587 \
  --to=celia@cockroachlabs.com \
  --qualify-bucket=foo \
  --qualify-object-prefix=foo \
  --release-bucket=foo \
  --release-object-prefix=foo
```

#### Test 2:
- Ran `post-blockers`, using fake parameters
- Got TODO message (expected): `TODO(celia): fetch blockers and send out 
```
SMTP_PASSWORD=foo \
GITHUB_TOKEN=foo \
bazel run pkg/cmd/release:release \
post-blockers -- \
  --dry-run \
  --release-series=21.1 \
  --smtp-user=foo \
  --smtp-host=smtp.gmail.com \
  --smtp-port=587 \
  --to=celia@cockroachlabs.com

```

Release note: None

rail reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76939#-:-MwbqDOSBAZEBr9OIQU9:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jlinder)



<!-- Sent from Reviewable.io -->

celiala commented:
Hmm, I'm not confident the vendor folder should be there (I think it was there before I created the branch). Let me investigate/verify.

celiala committed:
release: setup for post-blockers cmd

This commit does a quick refactor and sets up cobra command
infra for the post-blockers command.

Actual logic for fetching and posting blockers to an email
will happen in a follow-up commit.

Release note: None

celiala commented:
TFTR!

bors r=rail

celiala commented:
bors r=rail

rail reviewed (approved):
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76939)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @jlinder)



<!-- Sent from Reviewable.io -->

celiala commented:
bors r=rail
